sssd_test_framework/utils/ldb.py

```diff
@@ -0,0 +1,67 @@
+from __future__ import annotations
+
+from enum import Enum
+
+from pytest_mh import MultihostHost, MultihostUtility
+
+from sssd_test_framework.misc import parse_ldif
+
+
+class LDBUtils(MultihostUtility[MultihostHost]):
+    """
+    LDB Tools wrappper.
+    """
+
+    class Scope(Enum):
+        """
+        Search scope to use.
+        """
+
+        ONE_LEVEL = "one"
+        SUBTREE = "sub"
+        BASE = "base"
+
+    def search(
+        self,
+        path: str,
+        basedn: str | None = None,
+        scope: "LDBUtils.Scope" | None = None,
```

Comment:
Quotes can be removed when `from __future__ import annotations` is present.

Reply:
Default scope should be subtree. This is the default scope when calling ldbsearch without `-s`. Even though it is unlikely that the default value will change, we should hard code it here to avoid issues if it happens.